package.json

```diff
@@ -47,14 +47,23 @@
     "@types/node": "^22.7.5",
     "eslint": "^9.14.0",
     "eslint-config-prettier": "^9.1.0",
+    "husky": "^9.1.6",
     "jest": "^29.7.0",
     "jest-environment-jsdom": "^29.7.0",
+    "lint-staged": "^15.2.10",
     "prettier": "^3.3.3",
     "ts-jest": "^29.2.5",
     "typescript": "^5.6.3",
     "typescript-eslint": "^8.12.2"
   },
   "engines": {
     "node": ">=18"
+  },
+  "lint-staged": {
+    "*.{js,ts}": [
+      "prettier --write",
+      "eslint --fix"
+    ],
+    "!(*.js|*.ts)": "prettier --write --ignore-unknown"
```

Comment:
Is this one necessary? What's its purpose?

Reply:
Yes, this formats all other files that are being committed and [supported by Prettier](https://prettier.io/docs/en/) (e.g. YAML, Markdown, etc). Since ESLint as configured here only supports JavaScript and TypeScript we cannot add ESLint to this task.